src/include/self_driving/pilot/action/change_knob_action_generator.h

```diff
@@ -19,6 +19,7 @@ class AbstractAction;
  * Generate change knob candidate actions
  */
 class ChangeKnobActionGenerator {
```

Comment:
I feel like there should be an `AbstractGenerator` class. These can then go into a `generators` sub-directory.

Reply:
Sure! I'll do the refactor.